core/services/feeds/service.go

```diff
@@ -510,6 +510,21 @@ func (s *service) DeleteJob(ctx context.Context, args *DeleteJobArgs) (int64, er
 		logger.Errorw("Failed to push metrics for job proposal deletion", "err", err)
 	}
 
+	job, err := s.jobORM.FindJobByExternalJobID(ctx, proposal.ExternalJobID.UUID)
+	if err != nil {
+		// TODO: can this fail for non-workflow jobs? if so, should we still return an error (we don't know the type yet...)
+		return 0, errors.Wrap(err, "FindJobByExternalJobID failed")
+	}
+	if job.WorkflowSpecID != nil {
```

Comment:
I *think* if FindJobByExternalJobID errors with SqlNoRows the job is just Job{}? which would be fine here and skip this if

Reply:
But I still need to know if this is a workflow spec or some other type.